packages/autocomplete-plugin-recent-searches/src/createRecentSearchesPlugin.ts

```diff
@@ -1,114 +1,118 @@
-              icon.innerHTML = recentIcon;
-              const title = document.createElement('div');
-              title.className = 'title';
-              title.innerText = item.query;
-              leftItems.appendChild(icon);
-              leftItems.appendChild(title);
+    subscribed: {
+      onSelect({ item, state, source }) {
+        const inputValue = source.getItemInputValue({ item, state });
 
-              const removeButton = document.createElement('button');
-              removeButton.className = 'item-icon removeButton';
-              removeButton.type = 'button';
-              removeButton.innerHTML = resetIcon;
-              removeButton.title = 'Remove';
-
-              container.appendChild(leftItems);
-              container.appendChild(removeButton);
-              root.appendChild(container);
-
-              removeButton.addEventListener('click', (event) => {
-                event.stopPropagation();
-                store.remove(item);
-                refresh();
-              });
-            },
-          },
-        },
-      ];
+        if (inputValue) {
+          store.add({
+            id: inputValue,
+            query: inputValue,
+          } as TItem);
+        }
+      },
     },
-    onSubmit: ({ state }) => {
+    onSubmit({ state }) {
       const { query } = state;
+
       if (query) {
         store.add({
-          objectID: query,
+          id: query,
           query,
-        });
+        } as TItem);
       }
     },
-    onSelect: ({ item, state, source }) => {
-      const inputValue = source.getItemInputValue({ item, state });
-      const { objectID } = item as any;
-      if (inputValue) {
-        store.add({
-          objectID: objectID || inputValue,
-          query: inputValue,
-        });
-      }
+    getSources({ query, refresh }) {
+      lastItemsRef.current = store.getAll(query);
+
+      return Promise.resolve(lastItemsRef.current).then((items) => {
+        if (items.length === 0) {
+          return [];
+        }
+
+        return [
+          {
+            getItemInputValue({ item }) {
+              return item.query;
+            },
+            getItems() {
+              return items;
+            },
+            templates: getTemplates({
+              onRemove(id) {
+                store.remove(id);
+                refresh();
+              },
+            }),
+          },
+        ];
+      });
     },
     data: {
-      getFacetFilters: () => {
-        return store.getAll().map((item) => [`objectID:-${item.query}`]);
+      getQuerySuggestionsFacetFilters() {
+        // If the items returned by `store.getAll` are contained in a Promise,
+        // we cannot provide the facet filters in time when this function is called
+        // because we need to resolve the promise before getting the value.
+        if (!Array.isArray(lastItemsRef.current)) {
+          // @TODO: use the `warn` function from `autocomplete-core`
```

Comment:
I wonder if we could make an explicit `internal` folder/file to reexport internals from -core?

Reply:
I expect us to come up with a proper solution before releasing as well. I'm not too sure what would be the impact of having a shared package that is not published. We need to look into it. (But that's outside of this PR's scope.)